src/ILLink.RoslynAnalyzer/RequiresAnalyzerBase.cs

```diff
@@ -126,6 +134,28 @@ public override void Initialize (AnalysisContext context)
 						ReportRequiresDiagnostic (operationContext, member, requiresAttribute);
 					}
 				}
+
+				void CheckOverrides (
+					SymbolAnalysisContext symbolAnalysisContext,
+					IMethodSymbol method)
+				{
+					if (!method.HasAttribute (RequiresAttributeName)) {
+						while (method.OverriddenMethod != null) {
```

Comment:
Since this gets called for each method in the compilation, would it be enough to just check the immediate overridden method?

I think we also will need to look for matching annotations on implemented interface methods.

Reply:
Yeah, we should test things like:

```C#
using System.Diagnostics.CodeAnalysis;

interface IRUC
{
	[RequiresUnreferencedCode("RUC Message")]
	public void RUC();
}

class Imp : IRUC
{
	public void RUC()
	{
	}
}
```

Produce `IL2046` (in the analyzer).